### PR TITLE
Bring back Flake8 

### PR DESCRIPTION
**Pull request by squeaky-pl**
Some parts of the code hint me that there used to be Flake8 running on this code. For the sake of more safety let's bring that static analysis back. I brought back the latest version of Flake8

Problems that needed to be fixed:
- comparing `None` and `True` with `==` in sqlalchemy queries
- dead imports
- mixing tabs and spaces, missing new lines
- incorrect string escapes in regexps such as `"\s"` (needs to be either `"\\s"` or `r"\s"`)
- nondescriptive var names

One particularly annoying problem I had to solve is incorrectly using imports to make pytest fixtures visible instead of relying on coftests.py / plugins. This chokes flake8 because fixture such as `api_client` would be imported and then used as a param in test which shadows the original import. I moved fixtures to conftest.py.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -25,6 +25,25 @@ jobs:
           when: always
           command: isort -c .
 
+  static-code-analysis-flake8:
+    docker:
+      - image: circleci/python:2.7
```
Comment (squeaky-pl) [inline .circleci/config.yml:30]: Sadly I need a separate job with Python 2. This is of course interim solution, and hopefully would change in near future,

**Comment by AlecRosenbaum** [comment]
I like the direction, but there's too much going on here for me to feel comfortable merging in one go. I don't think that I'd catch anything that's subtly different among everything else.

Can you split this up into a few different changes that first introduce the flake8 checker ignoring everything that doesn't already pass (a long todo list of ignore codes within `.flake8` would be fine), then do different PR's for each or small groups of error codes?

**Review by AlecRosenbaum — COMMENTED** [review]
ignore -- just commenting since the above comment didn't count as a review and I don't want to get reminders

**Comment by squeaky-pl** [comment]
This was addressed in other PRs.

**Pull request closed without merging**